File: web_api/schemas/provider_status.py
from pydantic import BaseModel, Field


class ProviderState(BaseModel):
    configured: bool = False
    env_keys_present: list[str] = Field(default_factory=list)
    status: str = "config_missing"
    safe_display: dict[str, str] = Field(default_factory=dict)
    error_type: str | None = None


class ProviderStatusResponse(BaseModel):
    engine: str = "internal"
    no_send: bool = True
    providers: dict[str, ProviderState]
    warnings: list[str] = Field(default_factory=list)

File: web_frontend/src/api/providerStatus.ts
import { apiGet } from "./client";

export type ProviderState = {
  configured: boolean;
  env_keys_present: string[];
  status: "configured" | "config_missing" | "invalid_config" | "disabled" | string;
  safe_display: Record<string, string>;
  error_type?: string | null;
};

export type ProviderStatus = {
  engine: "internal";
  no_send: boolean;
  providers: Record<string, ProviderState>;
  warnings: string[];
};

export function getProviderStatus() {
  return apiGet<ProviderStatus>("/api/provider-status");
}
